projects/igniteui-angular/src/lib/grids/common/row.interface.ts

```diff
@@ -23,6 +27,7 @@ export interface RowType {
     children?: ITreeGridRecord[];
     parent?: ITreeGridRecord;
     hasChildren?: boolean;
+    grid: IgxGridComponent | IgxTreeGridComponent | IgxHierarchicalGridComponent;
```

Comment:
Can this just be a base grid?

Reply:
Sure, it can, the better option. And either way a dev should cast to the correct type if needed to:
```
public eventHandler(row: RowType) {
   const grupingExpr = (row.grid as IgxGridComponent).groupingExpressions;
}
```